docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +134,177 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
+
+See `Resolving the error` section for the first example on this page.
+
+
+## PrivateApi:DependencyInBusinessModel
+
+Business models on project level should avoid using private API from Core level
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessFilter` uses as dependency `CustomerAccessReaderInterface` that was declared on the vendor level.
+
+```php
+...
+class CustomerAccessFilter implements CustomCustomerAccessFilter
+{
+    /**
+    * @param \Pyz\Zed\CustomerAccess\CustomerAccessConfig $customerAccessConfig
+    * @param \Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface $customerAccessReader
+    */
+    public function __construct(CustomerAccessConfig $customerAccessConfig, CustomerAccessReaderInterface $customerAccessReader)
+    {
+        $this->customerAccessReader = $customerAccessReader;
+        $this->customerAccessConfig = $customerAccessConfig;
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+PrivateApi:DependencyInBusinessModel "Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter"
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+See `Resolving the error` section for the first example on this page.
+
+
+## PrivateApi:ObjectInitialization
+
+Business models on project level should avoid creating private API from Core level.
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessFilter` contains creating `CustomerAccessReader` in the `constructor` method.
+
+```php
+...
+class CustomerAccessFilter implements CustomCustomerAccessFilter
+{
+    public function __construct(
+        CustomerAccessEntityManagerInterface $customerAccessEntityManager,
+        CustomerAccessReaderInterface $customerAccessReader,
+        CustomerAccessFilterInterface $customerAccessFilter
+    ) {
+        parent::__construct($customerAccessEntityManager);
+        $this->customerAccessReader = new CustomerAccessReader();
+        $this->customerAccessFilter = $customerAccessFilter;
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+See `Resolving the error` section for the first example on this page.
```

Comment:
Provide the link on all places where you redirect the user of the documentation.

Text should be also changed:

```
Solution is the error is the same as in `Resolving the error`[#link-on-something]
```

Reply:
++